lib/qiniu/chunk_upload.ex

```diff
@@ -0,0 +1,137 @@
+defmodule Qiniu.ChunkUpload do
+  @moduledoc """
+  For chunked uploading.
+  """
+
+  alias Qiniu.PutPolicy
+
+  @default_block_size 1024 * 1024 * 4
+
+  @default_chunk_size 1024 * 1024
+
+
+  @doc """
+  Qiniu chunked upload:
+  https://developer.qiniu.com/kodo/manual/1650/chunked-upload
+
+  ## Options
+    * `:key`   - file name in a Qiniu bucket
+    * `:mimeType`   - file name in a Qiniu bucket
+  """
+  def chunk_upload(put_policy, local_file, opts \\ [])
+
+  def chunk_upload(%PutPolicy{}=put_policy, local_file, opts) do
+    uptoken = Qiniu.Auth.generate_uptoken(put_policy)
+    chunk_upload(uptoken, local_file, opts)
+  end
+
+  def chunk_upload(uptoken, local_file, opts) when is_binary(uptoken) do
+    case File.stat(local_file) do
+      {:ok, %{size: file_size, type: :regular}} ->
+        File.stream!(local_file, [], @default_block_size)
+        |> Stream.map(fn(blk) ->
```

Comment:
I think that anonymous function is  explicit enough:  `A block |> chop  |> upload`